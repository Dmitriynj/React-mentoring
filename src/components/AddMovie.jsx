import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Input, InputLabel, Button } from '@material-ui/core';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },
  confirmButton: {
    marginLeft: 10,
    background: '#f65261',
    color: 'white',
    borderRadius: 4,
    border: '2px solid #f65261',
    '&:hover': { background: '#f65261' },
  },
  resetButton: {
    borderRadius: 4,
    border: '2px solid #f65261',
    color: '#f65261',
  },
}));

const AddMovie = () => {
  const classes = useStyles();

  return (
    <>
      <section className={classes.field}>
        <InputLabel className={classes.label}>MOVIE ID</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>TITLE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>RELEASE DATE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>MOVIE URL</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>GENRE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>OVERVIEW</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel color="secondary" className={classes.label}>
          RUNTIME
        </InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.footer}>
        <Button className={classes.resetButton}>Reset</Button>
        <Button className={classes.confirmButton}>Submit</Button>
      </section>
    </>
  );
};

export { AddMovie };
