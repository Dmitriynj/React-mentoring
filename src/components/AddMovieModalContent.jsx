import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Input, InputLabel } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { createMovie } from '../store/thunks';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },

  ...MODAL_BUTTONS_STYLES,
}));

const AddNewMovieStateless = ({ sendCreateReq, onConfirm }) => {
  const classes = useStyles();
  const [creatingMovie, setCreatingMovie] = useState({
    title: '',
    release_date: '',
    overview: '',
    genres: [''],
    runtime: '',
    poster_path: '',
  });

  console.log('creating movie', creatingMovie);

  const onChangeInput = (event) => {
    const { name } = event.target;
    let { value } = event.target;
    if (name === 'runtime') {
      value = Number(value);
    }
    if (name === 'genres') {
      value = value.split(',');
    }
    setCreatingMovie({ ...creatingMovie, [name]: value });
  };

  const onSubmitChanges = () => {
    sendCreateReq(creatingMovie);
    onConfirm();
  };

  return (
    <>
      <section className={classes.field} key="title-field">
        <InputLabel className={classes.label}>TITLE</InputLabel>
        <Input
          name="title"
          type="text"
          onChange={onChangeInput}
          value={creatingMovie.title}
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="release-date-field">
        <InputLabel className={classes.label}>RELEASE DATE</InputLabel>
        <Input
          name="release_date"
          onChange={onChangeInput}
          value={creatingMovie.release_date}
          type="date"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="genres-field">
        <InputLabel className={classes.label}>GENRES</InputLabel>
        <Input
          name="genres"
          onChange={onChangeInput}
          value={creatingMovie.genres.join(',')}
          type="text"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="overview-field">
        <InputLabel className={classes.label}>OVERVIEW</InputLabel>
        <Input
          name="overview"
          onChange={onChangeInput}
          value={creatingMovie.overview}
          type="text"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="runtime-field">
        <InputLabel className={classes.label}>RUNTIME</InputLabel>
        <Input
          name="runtime"
          onChange={onChangeInput}
          value={creatingMovie.runtime}
          type="number"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="poster-path-field">
        <InputLabel className={classes.label}>POSTER PATH</InputLabel>
        <Input
          name="poster_path"
          onChange={onChangeInput}
          value={creatingMovie.poster_path}
          type="text"
          className={classes.input}
        />
      </section>

      <section className={classes.footer} key="footer">
        <Button onClick={onSubmitChanges} className={classes.confirmButton}>
          Create
        </Button>
      </section>
    </>
  );
};
AddNewMovieStateless.propTypes = {
  sendCreateReq: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  sendCreateReq: (data) => dispatch(createMovie(data)),
});

const AddMovieModalContent = connect(null, mapDispatchToProps)(AddNewMovieStateless);

export { AddMovieModalContent };
