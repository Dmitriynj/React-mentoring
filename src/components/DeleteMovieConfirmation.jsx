import React from 'react';
import { Typography, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  footer: { display: 'flex', justifyContent: 'flex-end', padding: 10 },

  ...MODAL_BUTTONS_STYLES,
}));

const DeleteMovieConfirmation = ({ onConfirm }) => {
  const classes = useStyles();

  return (
    <>
      <Typography>Are you really want to delete this movie?</Typography>
      <section className={classes.footer}>
        <Button onClick={onConfirm} className={classes.confirmButton}>
          Submit
        </Button>
      </section>
    </>
  );
};

DeleteMovieConfirmation.propTypes = {
  onConfirm: PropTypes.func,
};
DeleteMovieConfirmation.defaultProps = {
  onConfirm: () => {},
};

export { DeleteMovieConfirmation };
