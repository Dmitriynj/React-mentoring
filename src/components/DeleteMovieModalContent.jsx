import React from 'react';
import { Typography, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  footer: { display: 'flex', justifyContent: 'flex-end', padding: 10 },
  ...MODAL_BUTTONS_STYLES,
}));

const DeleteMovieModalContent = ({ onConfirm }) => {
  const classes = useStyles();

  return (
    <>
      <Typography>Are you really want to delete this movie?</Typography>
      <section className={classes.footer}>
        <Button onClick={onConfirm} className={classes.confirmButton}>
          Submit
        </Button>
      </section>
    </>
  );
};

DeleteMovieModalContent.propTypes = {
  onConfirm: PropTypes.func,
};
DeleteMovieModalContent.defaultProps = {
  onConfirm: () => {},
};

export { DeleteMovieModalContent };
