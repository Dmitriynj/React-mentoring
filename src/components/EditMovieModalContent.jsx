import React from 'react';
import { Button, Input, InputLabel } from '@material-ui/core';
import { uniqueId, map } from 'lodash';
import { makeStyles } from '@material-ui/core/styles';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },

  ...MODAL_BUTTONS_STYLES,
}));

const movieFields = [
  { name: 'TITLE' },
  { name: 'RELEASE DATE' },
  { name: 'MOVIE URL' },
  { name: 'GENRE' },
  { name: 'OVERVIEW' },
  { name: 'RUNTIME' },
];

const EditMovieModalContent = () => {
  const classes = useStyles();

  const fieldElement = ({ value, name }) => {
    return (
      <section className={classes.field} key={`field-${uniqueId()}`}>
        <InputLabel className={classes.label}>{name}</InputLabel>
        <Input defaultValue={value} type="text" className={classes.input} />
      </section>
    );
  };

  const fieldElements = map(movieFields, fieldElement);

  return (
    <>
      {fieldElements}
      <section className={classes.footer}>
        <Button className={classes.resetButton}>Reset</Button>
        <Button className={classes.confirmButton}>Submit</Button>
      </section>
    </>
  );
};

export { EditMovieModalContent };
