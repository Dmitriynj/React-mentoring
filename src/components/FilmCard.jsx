import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Card, CardMedia, CardHeader, Typography, CardContent } from '@material-ui/core';
import PropTypes from 'prop-types';
import { ManagedMovie } from './ManagedMovie';
import { useAppState } from '../hooks/useAppState';

const useStyles = makeStyles({
  card: {
    maxWidth: 220,
    backgroundColor: '#232323',
    color: '#eeeeee',
    margin: '5px auto',
  },
  cardHeaderTitle: {
    fontSize: 18,
  },
  wrapper: {
    position: 'relative',
  },
  moreContainer: {
    position: 'absolute',
    width: '100%',
    display: 'flex',
    justifyContent: 'flex-end',
  },
  more: {
    margin: 10,
    position: 'relative',
    backgroundColor: '#232323',
    borderRadius: '50%',
    color: 'white',
    height: 25,
    width: 25,
    '&:hover': {
      backgroundColor: '#232323',
    },
  },
  media: {
    cursor: 'pointer',
    height: 240,
  },
  actionButton: {
    color: 'white',
  },
  menuItem: {
    background: '#232323',
    color: 'white',
    '&:hover': {
      background: '#232323',
    },
  },
  description: {
    overflow: 'hidden',
    display: '-webkit-box',
    '-webkit-line-clamp': 3,
    '-webkit-box-orient': 'vertical',
  },
});

const FilmCard = ({ id, title, imageUrl, description, genres, releaseDate, avgVote, budget }) => {
  const classes = useStyles();
  const { setCurrentMovie } = useAppState();

  const openMovieDetails = () => {
    document.querySelector('#header-content').scrollIntoView({
      behavior: 'smooth',
    });
    setCurrentMovie({ title, imageUrl, description });
  };

  console.log('movie id', id);

  return (
    <Card className={classes.card}>
      <CardHeader
        classes={{ title: classes.cardHeaderTitle }}
        action={<ManagedMovie id={id} />}
        title={title}
      />
      <CardMedia
        className={classes.media}
        image={imageUrl}
        title="Contemplative Reptile"
        onClick={openMovieDetails}
      />
      <CardContent>
        <div className={classes.description}>
          <Typography variant="body2" component="p">
            {description}
          </Typography>
        </div>
        <Typography variant="body2" component="p">
          Genres: {genres.join(',')}
        </Typography>
        <Typography variant="body2" component="p">
          Release date: {releaseDate}
        </Typography>
        <Typography variant="body2" component="p">
          Avg vote: {avgVote}
        </Typography>
        <Typography variant="body2" component="p">
          Budget: {budget}
        </Typography>
      </CardContent>
    </Card>
  );
};

FilmCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  genres: PropTypes.array.isRequired,
  releaseDate: PropTypes.string.isRequired,
  avgVote: PropTypes.number,
  budget: PropTypes.number,
};
FilmCard.defaultProps = {
  imageUrl: '',
  avgVote: 0,
  budget: 0,
};

export { FilmCard };
