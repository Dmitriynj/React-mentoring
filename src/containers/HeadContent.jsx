import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import { isEmpty } from 'lodash';
import { useAppState } from '../hooks/useAppState';
import { Search } from '../components/Search';
import { MovieDetails } from '../components/MovieDetails';
import { changeQueryOptions } from '../store/actions';

const BACKGROUND_IMAGE_URL = 'background.jpg';

const useStyles = makeStyles(() => ({
  mainImage: {
    border: 0,
    height: '60vh',
    margin: 0,
    display: 'flex',
    padding: 0,
    overflow: 'hidden',
    position: 'relative',
    maxHeight: 1000,
    alignItems: 'center',
    backgroundSize: 'cover',
    backgroundPosition: 'center center',
    backgroundImage: `url(${BACKGROUND_IMAGE_URL} )`,
  },
  shadowed: {
    backgroundImage: `-webkit-gradient(linear,left bottom,left top,color-stop(50%,rgba(0,0,0,0)),to(rgba(0,0,0,.7))),radial-gradient(50% 100%,rgba(0,0,0,0) 50%,rgba(0,0,0,.7) 100%)`,
    width: '100%',
    height: '100%',
  },
}));

const HeadContentStateless = ({ updateQueryOption }) => {
  const classes = useStyles();
  const { currentMovie } = useAppState();

  const search = (value) => {
    updateQueryOption({
      search: value,
      searchBy: 'title',
    });
  };

  return (
    <div className={classes.mainImage}>
      <div className={classes.shadowed} id="header-content">
        {isEmpty(currentMovie) ? <Search onSearch={search} /> : <MovieDetails {...currentMovie} />}
      </div>
    </div>
  );
};
HeadContentStateless.propTypes = {
  updateQueryOption: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  updateQueryOption: (data) => dispatch(changeQueryOptions(data)),
});

const HeadContent = connect(null, mapDispatchToProps)(HeadContentStateless);

export { HeadContent };
