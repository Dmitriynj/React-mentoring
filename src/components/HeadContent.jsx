import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { MovieDetails } from '../containers/MovieDetails';
import { Search } from './Search';

const useStyles = makeStyles(() => ({
  mainImage: {
    border: 0,
    height: '60vh',
    margin: 0,
    display: 'flex',
    padding: 0,
    overflow: 'hidden',
    position: 'relative',
    maxHeight: 1000,
    alignItems: 'center',
    backgroundSize: 'cover',
    backgroundPosition: 'center center',
    backgroundImage: `url('/background.jpg')`,
  },
  shadowed: {
    backgroundImage: `-webkit-gradient(linear,left bottom,left top,color-stop(50%,rgba(0,0,0,0)),to(rgba(0,0,0,.7))),radial-gradient(50% 100%,rgba(0,0,0,0) 50%,rgba(0,0,0,.7) 100%)`,
    width: '100%',
    height: '100%',
  },
}));

const HeadContent = () => {
  const classes = useStyles();

  return (
    <div className={classes.mainImage}>
      <div className={classes.shadowed} id="header-content">
        <Switch>
          <Route exact path={['/movies', '/no-movies']}>
            <Search />
          </Route>
          <Route exact path={['/movie/:id', '/movies']}>
            <MovieDetails />
          </Route>
        </Switch>
      </div>
    </div>
  );
};

export { HeadContent };
