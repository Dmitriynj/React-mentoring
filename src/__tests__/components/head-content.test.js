import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render } from '../test-utils';
import { HeadContent } from '../../components/HeadContent';

function renderComponent() {
  return render(
    <MemoryRouter initialEntries={['/movies']}>
      <Route exact path="/movies">
        <HeadContent />
      </Route>
    </MemoryRouter>,
    { initialState: {} }
  );
}

describe('Head content', () => {
  it('renders correctly', () => {
    const { container } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
