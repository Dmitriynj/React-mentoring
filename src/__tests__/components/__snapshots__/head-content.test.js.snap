// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head content renders correctly 1`] = `
<div
  class="makeStyles-mainImage-1"
>
  <div
    class="makeStyles-shadowed-2"
    id="header-content"
  >
    <div
      class="makeStyles-root-3"
    >
      <h4
        class="MuiTypography-root makeStyles-title-4 MuiTypography-h4"
      >
        FIND YOUR MOVIE
      </h4>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-searchElement-5"
      >
        <input
          class="MuiInputBase-input MuiInput-input"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
